### DISCUSSION TITLE
Can I use the compiler on my own site and how?

### DISCUSSION THREAD

**Discussion by alkoumi** [hiukim/mind-ar-js#111] [Q&A]
Can I use the compiler on my own site and how? thanks for this great work

**Comment by davidbrandpro** [comment]
Yeah, I also noticed that the documentation doesn't mention anything about the use of the compiler.

How do we use it locally or on our server?

**Comment by SpudRat3** [comment]
If you are reluctant to publish the code, perhaps an API, where we could send a .json of our targets (specifying URI), and get back the .mind as a return package? Just a thought. Sure would be useful.

**Comment by larstel** [comment]
The [compiler](https://github.com/hiukim/mind-ar-js/blob/c39c9e4f4a289b924b1a90c6a3eb72fc8df990ca/src/image-target/compiler.js) is included in the source code. There is also a [wiki page](https://hiukim.github.io/mind-ar-js-doc/core-api/) on how to use the compiler but at the moment the page is "hidden" (not listed in navigation bar). I found the page by looking into the source code of the wiki. I tried it and was able to successfully compile an image.

**Reply by rialsa** [reply]
Hi @larstel , 

Thanks for sharing this, it has been very helpful for me. 

I did actually get to export the data from the compiler, but couldn't create a .mind file with the exported buffer.
Please, could you give me some advice in how to save the exported buffer to be used later as a target image?

**Reply by larstel** [reply]
take a look at this [sourcecode](https://github.com/hiukim/mind-ar-js-doc/blob/d13ad927647504852b0fa7aaf190a6893db4fa05/static/samples/compile.html) (especially lines 12-19) to get an idea of how that can work

**Reply by rialsa** [reply]
ok amazing thanks!

**Reply by Sergey55rex** [reply]
sorry, this code doesn't work right now

**Reply by jabbourdan** [reply]
Still not working ?
